### PR TITLE
Support builds against Qt6

### PR DESCRIPTION
**Pull request by equalsraf**
See https://github.com/equalsraf/neovim-qt/issues/839 for the list of issues

In terms of code this is ready for review. There are still runtime issues to be checked (font integer metrics and possibly imfocus). But since we continue to default to qt5, I think it is ok to proceed and handle those separately.

### CODE REVIEW

**Comment by equalsraf** [comment]
The replacement for QDesktopWidget via QWidget::windowHandle() segfaults.

**Comment by equalsraf** [comment]
Most builds look ok now, the Circle CI has a failed test but code wise it seems this branch builds for all qt5 and also qt6 on Mac OS (Azure pipeline). I dont think there are packages for qt6 yet.

Most of the changes are due to compatibility between the two versions. Time to start breaking this CR apart and get some of these into the master branch.

I did not get a chance to test this properly yet, the most concerning API changes in Qt6 are the fonts and the microfocus bit.

**Comment by jgehrig** [comment]
This is good to see! :+1: 

Let me know if I can help in any way: testing, code-review, bug-fixes, etc.

**Comment by equalsraf** [comment]
I'll break off the simple stuff into https://github.com/equalsraf/neovim-qt/pull/842

**Comment by equalsraf** [comment]
Everything else seems to be done, the remaining issue is the removal of forceintegermetrics, From the docs

> This style strategy has been deprecated since Qt 5.15.0. Use QFontMetrics to retrieve rounded font metrics.

I guess I need to hunt down for locations where QFontMetrics is not being used.

Attempting to run, I get a message "font reports bad fixed pitch metrics" for DejaVu Sans Mono, but i dont get the same message when running a build with qt5.

**Comment by equalsraf** [comment]
Failing to get a working CI for windows. The installed version of Qt seems to be missing the compatibility module.

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +19,7 @@ if(WIN32)
 	if (USE_STATIC_QT)
 		add_definitions(-DUSE_STATIC_QT)
 		link_directories(${CMAKE_PREFIX_PATH}/share/qt5/plugins/platforms)
-		set(QTLIBS ${QTLIBS} Qt5PlatformCompositorSupport qwindows)
+		set(QTLIBS ${QTLIBS} Qt5${QT_VERSION_MAJOR}latformCompositorSupport qwindows)
```
Comment (equalsraf) [inline src/gui/CMakeLists.txt:22]: This looks wrong - this would expand as `Qt56latformCompositorSupport`

**Comment by equalsraf** [comment]
Merging, on the basis that the default is Qt5 and can continue to work on pending items as needed. The following caveats remain:

- See https://github.com/equalsraf/neovim-qt/issues/839 for the list of remaining issues
- the appveyor job is failing and is allowed to fail until we can fix the one failing test

**Pull request merged**